spec/unit/azurerm_server_create_spec.rb

```diff
@@ -557,20 +555,44 @@
     end
 
     describe "create_network_profile" do
-      it "successfully returns network profile response" do
-        @platform = 'Linux'
-        expect(@service).to receive(
-          :create_virtual_network).and_return(
-            stub_virtual_network_create_response)
-        expect(@service).to receive(
-          :create_subnet).and_return(
-            stub_subnet_create_response)
-        expect(@service).to receive(
-          :create_network_interface).and_return(
-            stub_network_interface_create_response)
-        response = @service.create_network_profile(@params)
-        expect(response.network_interfaces).to_not be nil
-        expect(response.network_interfaces).to be_a(Array)
+      context 'For non existing vent and subnet' do
+        it 'successfully returns network profile response' do
+          allow(@network_client).to receive_message_chain(:virtual_networks, :get).and_return(@network_promise)
+          allow(@network_promise).to receive_message_chain(:value!, :body).and_return(nil)
```

Comment:
These 2 lines stub the `vnet_exist?` and `subnet_exist?` methods. Please add comment for the same.